addon/mixins/route.js

```diff
@@ -283,17 +278,61 @@ export default Ember.Mixin.create({
 
    @method updateInfinityModel
    @param {Ember.Enumerable} newObjects The new objects to add to the model
-   @return {Ember.Array} returns the updated infinity model
+   @return {DS.RecordArray} returns the updated infinity model
    */
   updateInfinityModel(newObjects) {
```

Comment:
`updateInfinityModel` is a public hook that current addon consumers are using to modify what happens when a payload is received.  

By moving all of this functionality here, we're introducing breaking changes to users who are using this hook, and we're making the hook less useable.

Ember Infinity had over 250,000 downloads on NPM last month - this refactor is fantastic, but for such a simple lib we really can't be introducing breaking changes!